### DISCUSSION TITLE
Need help on Ecar import failure

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#14] [Bugs] [answered]
Hello All,

I tried importing ecar in the mobile but it is failing.
![image](https://user-images.githubusercontent.com/65657373/170696633-66d4bbfa-e58d-4e3d-8d14-67d0e60b4f16.png)
How do we get to know the failure reason ?
Any logs that get generated ?
Or is there any way we can turn on the logs for us to understand ?
Can we make use of Android Studio for getting more information?

So that would help us fix ecar imports , can you please guide @swayangjit

cc: @Krishnaj20 @coolbung @baljeetsingh

**Comment by Xitija** [comment]
@sharathkashyap Sir,

Apologies for adding multiple issues to the same ticket .
I have created this new ticket for Import Ecar Issue.
I am opening a new ticket for the [UI issue](https://github.com/Sunbird-Ed/Community/discussions/15) as it is not yet resolved .

**Comment by swayangjit** [comment, marked as answer]
@Xitija Seems like in the ECAR file name there is an extra "."(dot) . So it's breaking the import process. Please try to rename the package by removing the extra "."(dot)  and then import.

**Reply by Xitija** [reply]
'do_11345176856416256012.ecar' is the name of the ecar I am trying to import .
Attaching ecar here in zip format.
[do_11345176856416256012.zip](https://github.com/Sunbird-Ed/Community/files/8786712/do_11345176856416256012.zip)

**Reply by swayangjit** [reply]
@Xitija Where did you get this ecar ?

**Reply by Xitija** [reply]
We have created this new Ecar manually as per the structure decided.

**Reply by swayangjit** [reply]
@Xitija  Ty [this ecar](https://drive.google.com/file/d/1JqFLZKd6JpWKj-H6umahiCm05dkO2PGL/view?usp=sharing) and  let me know if its working or not.

**Reply by Xitija** [reply]
@swayangjit Sir ,
The Ecar that you provided above got imported in the staging app but not as an entire Interactive content  but each question-set seperately 
![image](https://user-images.githubusercontent.com/65657373/170922444-a3954397-22c0-4154-b7bf-6fdd57889026.png)
Each Questionset is getting played as a seperate content and video is not even visible . 
It is not working as an Interactive Video as a whole.

**Reply by swayangjit** [reply]
@Xitija  Can you please try to download directly from the app instead of manipulating the  ecar and let us know if it's working or not.

**Reply by Xitija** [reply]
@swayangjit Sir ,
Directly downloading the Ecar (I am doing it by pressing download button under the interactive content) only dowloads the video and no question set as publish pipline is not yet generated for Interactive video containing Video and Question set both to be a part of a single content . But the Ecar you gave me works on the local Video player , but not on the mobile . 
We are trying manually created Ecars for now because publish pipeline work is still in progress.

**Reply by swayangjit** [reply]
@Xitija It seems the zip file that you have sent might have an issue. Let me check  and get back to you.

**Reply by swayangjit** [reply]
@Xitija  First of all the ECAR structure is completely wrong, all the 3 contents added to the item array have visibility as Default. Only root content should have visibility as default and others should have visibility as Parent. Also, to support this type of ECAR you have to do some modifications in the import process where video contents(having interception points) should be treated as a collection. You have to modify the import process in [this method](https://github.com/Sunbird-Ed/sunbird-mobile-sdk/blob/45e1775ffcea6c04cbba504fe958eaf75decf6f6/src/content/impl/content-service-impl.ts#L441).

**Reply by Xitija** [reply]
Can you please give any inputs on this ? , @itsvick 
cc : @baljeetsingh
![image](https://user-images.githubusercontent.com/65657373/170995300-004deb9f-6e57-482a-bc60-bd5832dc1e17.png)

 I checked the filesystem to see how Ecar is imported , all the question-sets are coming at the top level actually should go inside do_....6012 which is the parent , will changing the visibility keep the question set folder inside the parent? 
Also , images in question sets are not getting imported .
@swayangjit Sir

**Reply by Xitija** [reply]
I tried to set "visibility": "Parent" for question-set in the items array but then the Ecar fails to be imported in the mobile.

**Reply by sharathkashyap** [reply]
@Xitija Kindly let us know if the issue is resolved? Also kindly provide the solution if in case fix was applied

**Reply by Xitija** [reply]
@sharathkashyap Sir,
We have kept this feature on hold now
So we can close as of now .
